x-pack/plugins/ml/public/application/explorer/explorer_charts/explorer_charts_container.js

```diff
@@ -55,13 +55,25 @@ function getChartId(series) {
 
 // Wrapper for a single explorer chart
 function ExplorerChartContainer({ series, severity, tooManyBuckets, wrapLabel, mlUrlGenerator }) {
-  const redirectToSingleMetricViewer = useCallback(async () => {
-    const singleMetricViewerLink = await getExploreSeriesLink(mlUrlGenerator, series);
-    addItemToRecentlyAccessed('timeseriesexplorer', series.jobId, singleMetricViewerLink);
-
-    window.open(singleMetricViewerLink, '_blank');
+  const [explorerSeriesLink, setExplorerSeriesLink] = useState();
+
+  useEffect(() => {
+    let isCancelled = false;
+    const generateLink = async () => {
+      const singleMetricViewerLink = await getExploreSeriesLink(mlUrlGenerator, series);
+      if (!isCancelled) {
+        setExplorerSeriesLink(singleMetricViewerLink);
+      }
+    };
+    generateLink();
+    return () => {
+      isCancelled = true;
+    };
```

Comment:
I've seen this pattern multiple times, so for consistency, it's worth creating a custom hook, `useSafeAsyncState` maybe?

Reply:
I'll make a follow up for this in 7.12 to consolidate all the places that are currently using this pattern. Since this is a bug fix, I'll keep this here for now.